nio/block/mixins/persistence/persistence.py

```diff
@@ -47,24 +47,40 @@ def _load(self):
         """
         self.logger.debug("Loading from persistence")
         # load whole item from persistence
-        item = self._persistence.load(self.id(), default={})
-        for persisted_var in self.persisted_values():
-            if persisted_var in item:
-                self.logger.debug("Loaded value {} for attribute {}".format(
-                    item[persisted_var], persisted_var))
-                # Set the loaded value to the attribute on this class
-                setattr(self, persisted_var, item[persisted_var])
+        data = self._persistence.load(self.id(), default={})
+        if not data:
+            return
```

Comment:
Question here: Should we really not call `deserialize` even if there was no data saved? Is there a chance that the user would want to take some kind of different action in their deserialize method if no data was saved? Maybe load a reasonable default?

Reply:
I had a similar thought but decided it would be more error prone to call with an empty dictionary, I normally see `deserialize` called when there is data.